src/OpenLoco/Paint/PaintMiscEntity.cpp

```diff
@@ -0,0 +1,163 @@
+#include "PaintMiscEntity.h"
+#include "../CompanyManager.h"
+#include "../Graphics/Gfx.h"
+#include "../Interop/Interop.hpp"
+#include "../Localisation/StringIds.h"
+#include "../Map/Tile.h"
+#include "Paint.h"
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::Paint
+{
+    // 0x00440325
+    void paintMiscEntity(PaintSession& session, MiscBase* base)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        int8_t* wiggleYOffsets = reinterpret_cast<int8_t*>(0x4FAAC8);
+
+        switch (base->getSubType()) // 00440326-0044032A
+        {
+            case MiscThingType::exhaust: // 0
```

Comment:
Could you split each of the switch cases into a separate function. In the very very far future these will become virtual functions of the entity type.